src/TableFunctions/TableFunctionRemote.cpp

```diff
@@ -211,7 +211,16 @@ void TableFunctionRemote::parseArguments(const ASTPtr & ast_function, ContextPtr
         }
 
         if (arg_num < args.size())
-            throw Exception(help_message, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        {
+            if (is_cluster_function)
+            {
+                throw Exception(help_message, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+            }
+            else
+            {
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "user and password should be string literal (in single quotes)");
```

Comment:
I'd rather make it a bit differently. Above in line 206
```
            if (arg_num < args.size() && !sharding_key)
            {
                sharding_key = args[arg_num];
                ++arg_num;
            }
```
can be changed to 
```
            if (arg_num < args.size())
            {
                if (sharding_key)
                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "User name and password should be string literals (in single quotes)");
                sharding_key = args[arg_num];
                ++arg_num;
            }
```
because your fix will make incorrect exception message in case user and password were correctly quoted, but the user has added some extra arguments after `sharding_key` argument (in this case it will be correct to print exception message `throw Exception(help_message, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);` instead of yours).